api/context.go

```diff
@@ -234,6 +238,47 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (cw *CorsWrapper) ServeHTTP(
+	w http.ResponseWriter,
+	r *http.Request) {
+	allowedMethods := []string{
```

Comment:
Can we move this out of the function or inside the if stmt?  since it's static.  Same for allowed Headers.